src/makeCalculator.js

```diff
@@ -4,7 +4,31 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add: (currentResult) => (value) => currentResult + value,
+
+    subtract: (currentResult) => (value) => currentResult - value,
+
+    multiply: (currentResult) => (value) => currentResult * value,
+
+    divide: (valueResult) => (value) => {
+      return value === 0 ? 'Error' : valueResult / value;
+    },
+
+    operate: function (callback, value) {
+      this.result = callback(this.result)(value);
+
+      return this;
+    },
```

Comment:
```suggestion
    operate(callback, value) {
      callback.call(this, value);
      
      return this;
    },
```

Reply:
You read more about a call method here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call

Reply:
this method I did not meet in training videos, thanks for the advice

Reply:
this method I did not meet in training videos, thanks for the advice